### PR TITLE
Provide CodeReport to code viewer dialog

### PR DESCRIPTION
**Pull request by beckerhe**
This is needed to get samples counts in the code viewer.

### CODE REVIEW

File: src/CodeViewer/Viewer.cpp [hunk 1]
```diff
@@ -100,10 +100,14 @@ void Viewer::DrawLineNumbers(QPaintEvent* event) {
        block = block.next()) {
     if (!block.isVisible() || bottom_of(block) < event->rect().top()) continue;
 
-    if (heatmap_bar_width_.Value() > 0.0 && heatmap_source_) {
+    if (heatmap_bar_width_.Value() > 0.0 && code_report_ != nullptr) {
```
Comment (karupayun) [inline src/CodeViewer/Viewer.cpp:103]: I would split these 2 conditions.

Here only if (code_report_ != nullptr) and the width of the heatmap before calculating intensity, what do you think? 

BTW, as we can skip conflicts between our PRs, I can change it in my PR. I'm currently thinking how to mix it when drawing the samples column.

**Review by karupayun — APPROVED** [review]
LGTM Henning! :)

File: src/CodeViewer/Viewer.cpp (hunk 1 above)
Comment (beckerhe) [reply src/CodeViewer/Viewer.cpp:103]: I don't understand. There is no work that I can do with either of these conditions not true.

Comment (beckerhe) [reply src/CodeViewer/Viewer.cpp:103]: What code should be executed when code_report != nullptr, but heatmap_bar_width.Value() == 0?

Comment (beckerhe) [reply src/CodeViewer/Viewer.cpp:103]: As discussed OOB, we will follow up tomorrow and I'm going to merge this as-is.

Comment (karupayun) [reply src/CodeViewer/Viewer.cpp:103]: Maybe we could want to hide the heatmap when hiding line numbers but letting the samples in?

**Pull request merged**